Dockerfile

```diff
@@ -1,23 +1,18 @@
 FROM kbase/sdkbase2:python
-MAINTAINER KBase Developer
-# -----------------------------------------
-
-# Insert apt-get instructions here to install
-# any required dependencies for your module.
-
-
-# -----------------------------------------
-
-# update installed WS client (will now include get_objects2)
-
-COPY ./ /kb/module
-RUN mkdir -p /kb/module/work
-RUN chmod -R a+rw /kb/module
+LABEL MAINTAINER KBase Developer
 
+COPY ./ /kb/module/
+ENV PYTHONPATH="/kb/module/lib:$PYTHONPATH"
 WORKDIR /kb/module
-
-RUN make all
+# install python modules one at a time so that all deps get resolved properly
+RUN pip install --upgrade pip && \
+    cat requirements.txt | sed -e '/^\s*#.*$/d' -e '/^\s*$/d' | xargs -n 1 pip install && \
+    cat requirements-test.txt | sed -e '/^\s*#.*$/d' -e '/^\s*$/d' | xargs -n 1 pip install && \
+    mkdir -p /kb/module/work && \
+    chmod -R a+rw /kb/module && \
+    cp compile_report.json work/ && \
+    make set-executable
```

Comment:
instead of needing kb-sdk on the base image, pre-compile the compile_report (and Impl/Server files) and transfer them over to the image. All the scripts are already in the repo, so instead of needing to run `make all` to create the scripts and compile, we just need to ensure the scripts are all executable.